addons/onboarding/static/src/views/form/onboarding_step_form_controller.js

```diff
@@ -20,6 +20,10 @@ export default class OnboardingStepFormController extends FormController {
     async save({ closable, ...otherParams }) {
         const saved = await super.save(otherParams);
         if (saved) {
+            var validated = await this.validateStepCondition();
+            if (!validated) {
+                return saved;
+            }
```

Comment:
Unless you use `validated` (which should probably be `isValidationConditionMet` or `toValidate`) further down as well, you could skip the variable.

```suggestion
            if (!await this.stepIsValidationConditionMet()) {
                return saved;
            }
```

Reply:
Oh, I think the CI test fails because you are not closing the modal with this early return.

Perhaps this works?
```js
        if (saved) {
            let shouldClose = closable;
            if (await this.stepIsValidationConditionMet()) {
                const { reloadOnFirstValidation, reloadAlways } = this.stepConfig;
                const validationResponse = await this.orm.call(
                    'onboarding.onboarding.step',
                    'action_validate_step',
                    [this.stepName],
                );
                if (reloadAlways || (reloadOnFirstValidation && validationResponse === "JUST_DONE")) {
                    shouldClose = false;
                    this.action.restore(this.action.currentController.jsId);
            } 
            if (shouldClose) {
                this.action.doAction({ type: "ir.actions.act_window_close" });
            }
        }
        return saved;
```